std/typecons.d

```diff
@@ -7838,3 +7838,210 @@ unittest
     assert(~Ternary.no == Ternary.yes);
     assert(~Ternary.unknown == Ternary.unknown);
 }
+
+/**
+Provides untyped storage of size `Size` aligned to `Alignment`.
+
+`Alignment` must be a power of 2.
+Default `Alignment` is the maximum alignment requirement on the current platform.
+
+Note: while this type ensures storage alignment on its own, it still obeys
+language rules. Therefore, care must be taken when using it in aggregate types,
+notably with custom `align` specification.
+
+Since aggregated object cannot test its alignment in an aggregate
+at compile time, an assert will be issued at runtime when a misaligned
+AlignedStorage is used.
+
+Example:
+
+    struct A
+    {
+        bool flag;
+        AlignedStorage!(4,4) storage; // OK, storage will be aligned to 4 bytes
+    }
+
+    struct B
+    {
+        align(1):
+        bool flag;
+        AlignedStorage!(4,4) storage; // WARNING: storage will be aligned to 1 byte
+    }
+
+    A a;
+    B b;
+    import std.conv : emplace;
+    emplace(cast(int*)(a.storage.ptr), 42); // OK
+    emplace(cast(int*)(b.storage.ptr), 42); // an AssertError will be thrown
+*/
+struct AlignedStorage(size_t Size, size_t Alignment = platformAlignment)
+{
+    /// Size of storage, in bytes
+    enum size = Size;
+
+    /// Alignment of storage, in bytes
+    enum alignment = Alignment;
+
+    @system @nogc nothrow:
+
+    /// Returns a pointer to the beginning of storage
+    @property inout(void)* ptr() inout
```

Comment:
This is not necessary. People can just take the address of the object.

Reply:
This goes through the invariant, asserting if alignment is out of whack.